src/test/regress/expected/start_stop_metadata_sync.out

```diff
@@ -0,0 +1,244 @@
+SET citus.next_shard_id TO 980000;
+SET client_min_messages TO WARNING;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+-- create a custom type for testing with a distributed table
+CREATE TYPE tt2 AS ENUM ('a', 'b');
+-- create test tables
+CREATE TABLE distributed_table_1(col int unique, b tt2);
+CREATE TABLE "distributed_table_2'! ?._"(col int unique);
+CREATE TABLE distributed_table_3(col int);
+CREATE TABLE distributed_table_4(a int UNIQUE NOT NULL, b int, c int);
+CREATE TABLE reference_table_1(col int unique);
+CREATE TABLE reference_table_2(col int unique);
+CREATE TABLE local_table(col int unique);
+-- create a fkey graph: dist -> dist -> ref1 <- local  && ref1 -> ref2
+ALTER TABLE distributed_table_1 ADD CONSTRAINT fkey_1 FOREIGN KEY (col) REFERENCES "distributed_table_2'! ?._"(col);
+ALTER TABLE "distributed_table_2'! ?._" ADD CONSTRAINT fkey_1 FOREIGN KEY (col) REFERENCES reference_table_1(col);
+ALTER TABLE reference_table_1 ADD CONSTRAINT fkey_1 FOREIGN KEY (col) REFERENCES reference_table_2(col);
+ALTER TABLE local_table ADD CONSTRAINT fkey_1 FOREIGN KEY (col) REFERENCES reference_table_1(col);
+SELECT create_reference_table('reference_table_2');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_reference_table('reference_table_1');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('"distributed_table_2''! ?._"', 'col');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('distributed_table_1', 'col');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('distributed_table_3', 'col');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('distributed_table_4', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE INDEX ind1 ON distributed_table_4(a);
+CREATE INDEX ind2 ON distributed_table_4(b);
+CREATE INDEX ind3 ON distributed_table_4(a, b);
+CREATE STATISTICS stat ON a,b FROM distributed_table_4;
+-- create views to make sure that they'll continue working after stop_sync
+INSERT INTO distributed_table_3 VALUES (1);
+CREATE VIEW test_view AS SELECT COUNT(*) FROM distributed_table_3;
+CREATE MATERIALIZED VIEW test_matview AS SELECT COUNT(*) FROM distributed_table_3;
+ALTER TABLE distributed_table_4 DROP COLUMN c;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+ start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+\c - - - :worker_1_port
+SET search_path TO "start_stop_metadata_sync";
+\d+ distributed_table_4
```

Comment:
I think we don't necessary need to show the output of `\d`, I'm pretty sure that in other tests we already ensure the existence of these constraints.

In this test, the important thing is that stop/start metadata sync with succeed with all these constraints.